File: backend/analysis/sentiment.py
from textblob import TextBlob
from django.http import JsonResponse

def analyze_sentiment(request):
    if request.method == 'POST':
        text = request.POST.get('text')
        blob = TextBlob(text)
        polarity = blob.sentiment.polarity
        subjectivity = blob.sentiment.subjectivity
        
        sentiment = "neutral"
        if polarity > 0:
            sentiment = "positive"
        elif polarity < 0:
            sentiment = "negative"
        
        result = {
            'text': text,
            'sentiment': sentiment,
            'polarity': polarity,
            'subjectivity': subjectivity,
        }
        return JsonResponse(result)
    return JsonResponse({'error': 'Invalid request method'}, status=405)

File: frontend2/pages/api/sentiment.js
import { TextBlob } from 'textblob';

export default async (req, res) => {
  if (req.method === 'POST') {
    try {
      const { text } = req.body;
      const analysis = new TextBlob(text);
      const sentiment = analysis.sentiment.polarity;
      res.status(200).json({ sentiment });
    } catch (error) {
      console.error('Sentiment analysis failed:', error);
      res.status(500).json({ message: 'Sentiment analysis failed.' });
    }
  } else {
    res.status(405).end(); 
  }
};